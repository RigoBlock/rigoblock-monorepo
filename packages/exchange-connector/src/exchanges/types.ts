export enum OrderType {
  BUY,
  SELL
}

export type Token = {
  address: string
  symbol: string
  decimal: number
  type: 'default'
}

export type OrdersList = Order[]

export type TickersList = Ticker[]

export type Order = {
  type: OrderType
  price: string
  amount: string
  // Number of orders at this price in case the order is an aggregation
  ordersCount?: number
}

export type Ticker = {
  priceEth: string
  priceUsd: string
  symbol: string
}

export type ExchangesMap<T> = {
  [key: string]: T
}

export interface IExchange<
  RawOrder = {},
  RawTicker = {},
  TICKERS_TOKEN_PAIRS = {}
> {
  networkId: number
  API_URL: string

  raw: {
    getOrders(
      baseToken: string,
      quoteToken: string,
      precision?: string
    ): Promise<RawOrder[] | RawOrder>

    getTickers(options: {
      networkId?: string
      granularity?: string
      tokenPairs?: TICKERS_TOKEN_PAIRS
    }): Promise<RawTicker[]>
  }

  getOrders(baseToken: string, quoteToken: string): Promise<OrdersList>

  getTickers(options: {
    networkId?: string
    granularity?: string
    tokenPairs?: TICKERS_TOKEN_PAIRS
  }): Promise<TickersList>

  network(id: number): IExchange
}
