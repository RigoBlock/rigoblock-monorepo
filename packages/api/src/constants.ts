export enum NETWORKS {
  MAINNET = '1',
  ROPSTEN = '3',
  KOVAN = '42',
  LOCALHOST = '5777'
}

export const RPC_URLS = {
  [NETWORKS.MAINNET]: 'wss://mainnet.infura.io/ws/',
  [NETWORKS.ROPSTEN]: 'wss://ropsten.infura.io/ws/',
  [NETWORKS.KOVAN]: 'wss://kovan.infura.io/ws/',
  [NETWORKS.LOCALHOST]: 'ws://localhost:8545'
}
