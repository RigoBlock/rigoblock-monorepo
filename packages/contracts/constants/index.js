const path = require('path')

const NETWORKS = {
  ganache: 'http://localhost:8545',
  mainnet: 'https://mainnet.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  ropsten: 'https://ropsten.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  kovan: 'https://kovan.infura.io/v3/d48872aa1c00471c825e9d856c3c3138'
}

module.exports = {
  TMP_DIR: '.tmp',
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../src'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'AEthfinex',
    'ATotlePrimary',
    'AWeth',
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Exchange',
    'ExchangeEfx',
    'ExchangeV1Fork',
    'ExchangesAuthority',
    'Faucet',
    'HGetDragoData',
    'Inflation',
    'Migrations',
    'NavVerifier',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'SigVerifier',
    'TokenTransferProxy',
    'TotlePrimary',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WETH9',
    'WrapperLock',
    'WrapperLockEth',
    'ZeroExExchangeHandler'
  ]
}
