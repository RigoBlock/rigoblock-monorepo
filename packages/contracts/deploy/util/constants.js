export const constants = {
    optimizerEnabled: false,
    timeoutMs: 30000,
    contracts: '*' as '*'
}
