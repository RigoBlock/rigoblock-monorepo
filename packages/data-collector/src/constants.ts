export const REDIS_HOST = process.env.REDIS_HOST || '127.0.0.1'
export const REDIS_PORT = parseInt(process.env.REDIS_PORT) || 6379
export const REDIS_PASSWORD = process.env.REDIS_PASSWORD || ''
export const TELEGRAF_HOST = process.env.TELEGRAF_HOST || '127.0.0.1'
export const TELEGRAF_PORT = process.env.TELEGRAF_PORT || 8125

export const TOKEN_MARKET_BASE_URL =
  'https://tokenmarket.net/blockchain/ethereum/assets'
export const CRYPTO_NEWS_BASE_URL =
  'https://cryptopanic.com/api/posts/?auth_token=5b3f861c261e7e6243a55dc7b7ca600832156886&public=true&filter=hot&currencies='
