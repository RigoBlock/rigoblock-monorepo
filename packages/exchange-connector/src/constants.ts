export const MAINNET: string = 'mainnet'
export const ROPSTEN: string = 'ropsten'
export const KOVAN: string = 'kovan'
export const RINKEBY: string = 'rinkeby'

export const PRICE_PRECISION: number = 7
export const AMOUNT_PRECISION: number = 5
export const TO_WEI: number = 1e18

export enum supportedExchanges {
  ETHFINEX = 'Ethfinex',
  ETHFINEX_RAW = 'EthfinexRaw',
  ERCDEX = 'ERCdEX',
  ZEROEXRELAYER = '0xStandardRelayer'
}

export enum NETWORKS {
  MAINNET = 1,
  KOVAN = 42,
  ROPSTEN = 3,
  RINKEBY = 4
}

export enum WS_STATUS {
  CLOSED = 'closed',
  OPEN = 'open',
  CONNECTING = 'connecting',
  CLOSING = 'closing'
}
