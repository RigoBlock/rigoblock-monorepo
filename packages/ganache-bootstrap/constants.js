module.exports = {
  GANACHE_URL: 'http://localhost:8545'
}
