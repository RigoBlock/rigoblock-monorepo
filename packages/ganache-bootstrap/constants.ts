const task = {
  GANACHE_URL: 'http://localhost:8545'
}

export default task
