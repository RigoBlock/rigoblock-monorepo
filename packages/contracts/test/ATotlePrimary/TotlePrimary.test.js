import { GANACHE_NETWORK_ID, GAS_ESTIMATE } from '../../constants'
import totlePrimaryArtifact from '../../artifacts/TotlePrimary.json'
import exchangeArtifact from '../../artifacts/Exchange.json'
import {
    assetDataUtils,
    BigNumber,
    ContractWrappers,
    generatePseudoRandomSalt,
    Order,
    orderHashUtils,
    signatureUtils,
    SignerType,
} from '0x.js'
import { ECSignature, SignatureType, SignedOrder, ValidatorSignature } from '@0x/types';
import web3 from '../web3'

const contractName = 'TotlePrimary'

describeContract(contractName, () => {
  let totlePrimaryAddress
  let totlePrimaryInstance
  let transactionDefault
  let exchangeAddress
  let exchangeInstance
  let rigoTokenAddress
  let weth9Address
  let erc20ProxyAddress
  let tokenTransferProxyAddress
  let zeroExHandlerAddress = baseContracts['ZeroExExchangeHandler'].address

  beforeAll(async () => {
    rigoTokenAddress = await baseContracts['RigoToken'].address
    weth9Address = await baseContracts['WETH9'].address
    erc20ProxyAddress = await baseContracts['Erc20Proxy'].address
    tokenTransferProxyAddress= await baseContracts['TokenTransferProxy'].address
    zeroExHandlerAddress = await baseContracts['ZeroExExchangeHandler'].address
    totlePrimaryAddress = await baseContracts[
      'TotlePrimary'
    ].address
    totlePrimaryInstance = new web3.eth.Contract(
      totlePrimaryArtifact.networks[GANACHE_NETWORK_ID].abi,
      totlePrimaryAddress
    )
    exchangeAddress = await baseContracts[
      'Exchange'
    ].address
    exchangeInstance = new web3.eth.Contract(
      exchangeArtifact.networks[GANACHE_NETWORK_ID].abi,
      exchangeAddress
    )
    transactionDefault = {
      from: accounts[0],
      gas: GAS_ESTIMATE,
      gasPrice: 1
    }
  })

  describe('performRebalance', () => {
    it('performs a 0x GRG buy order on totle', async () => {
      // account 1 buys from account 0
      const makerAddress = accounts[0]
      const takerAddress = '0x0000000000000000000000000000000000000000'
      const senderAddress = '0x0000000000000000000000000000000000000000'
      const feeRecipientAddress = '0x0000000000000000000000000000000000000000'
      const makerAssetData = assetDataUtils.encodeERC20AssetData(rigoTokenAddress)
      const takerAssetData = assetDataUtils.encodeERC20AssetData(weth9TokenAddress)
      //const exchangeAddress = exchangeAddress //'0x1d8643aae25841322ecde826862a9fa922770981'
      const salt = generatePseudoRandomSalt().toString()
      const makerFee = new BigNumber(0).toString()
      const takerFee = new BigNumber(0).toString()
      const makerAssetAmount = web3.utils.toWei('0.3').toString() // test with equal amounts (prev 0.2)
      const takerAssetAmount = web3.utils.toWei('0.3').toString()
      const expirationTimeSeconds = new BigNumber(
        Date.now() + 3600000
      ).toString() // Valid for up to an hour

      // default account sets allowance to 0x erc20proxy
      await baseContracts['RigoToken'].approve(erc20ProxyAddress, makerAssetAmount)

      // Generate order
      const order = {
        makerAddress,
        takerAddress,
        feeRecipientAddress,
        senderAddress,
        makerAssetAmount,
        takerAssetAmount,
        makerFee,
        takerFee,
        expirationTimeSeconds,
        salt,
        makerAssetData,
        takerAssetData,
        exchangeAddress
      }
      const providerEngine = web3.currentProvider

      const signerAddress = accounts[1]

      const signedOrder = await signatureUtils.ecSignOrderAsync(
        providerEngine,
        order,
        signerAddress
      )
      console.log(signedOrder)

      const takerAssetFillAmount = (takerAssetAmount / 2).toString() // partial fill

      const encodedOrder = await web3.eth.abi.encodeParameters(
        [
          'tuple(address,address,address,address,uint256,uint256,uint256,uint256,uint256,uint256,bytes,bytes)',
          'uint256',
          'bytes'
        ],
        [
          [
            makerAddress,takerAddress,feeRecipientAddress,senderAddress,
            makerAssetAmount,takerAssetAmount,makerFee,takerFee,expirationTimeSeconds,salt,
            makerAssetData,takerAssetData
          ],
          takerAssetFillAmount,
          signedOrder.signature
        ]
      )

      const tradeId = '0xfa39c1a29cab1aa241b62c2fd067a6602a9893c2afe09aaea371609e11cbd92d' // mock id bytes32
      const isSell = false // buying a token from the secondary account -> isSell = false
      const optionalTrade = false
      const tokenAmount = takerAssetFillAmount
      const minimumExchangeRate = 1 // Allowable Price Change (%)
      const minimumAcceptableTokenAmount = 10000 // Minimum Token Fill Quantity(%)

      const feeAccount = accounts[2]
      // TODO: check if deployed
      /*
      if(!affiliateRegistry.isValidAffiliate(feeAccount)){
          feeAccount = defaultFeeAccount;
      }
      */

      // accounts[1] takes the order, purchases GRG
      const transactionDetails = {
        value: takerAssetFillAmount,
        from: accounts[1],
        gas: GAS_ESTIMATE,
        gasPrice: 1
      }

      await expect(totlePrimaryInstance.methods.performRebalance(
        [
          [
            isSell,
            rigoTokenAddress,
            tokenAmount,
            optionalTrade,
            minimumExchangeRate, // check on value
            minimumAcceptableTokenAmount,
            [
                [
                  zeroExHandlerAddress,
                  encodedOrder
                ]
            ]
          ]
        ],
        feeAccount,
        tradeId
      ).send({ ...transactionDetails })
      ).rejects.toThrowErrorMatchingSnapshot()
    })
    it('performs a 0x GRG sell order on totle', async () => {
      //account 1 signs GRG buy order
      //must wrap eth
      const makerAddress = accounts[1]
      const takerAddress = '0x0000000000000000000000000000000000000000'
      const senderAddress = '0x0000000000000000000000000000000000000000'
      const feeRecipientAddress = '0x0000000000000000000000000000000000000000'
      const makerAssetData = assetDataUtils.encodeERC20AssetData(weth9TokenAddress)
      const takerAssetData = assetDataUtils.encodeERC20AssetData(rigoTokenAddress)
      //const exchangeAddress = exchangeAddress //'0x1d8643aae25841322ecde826862a9fa922770981'
      const salt = generatePseudoRandomSalt().toString()
      const makerFee = new BigNumber(0).toString()
      const takerFee = new BigNumber(0).toString()
      const makerAssetAmount = web3.utils.toWei('0.3').toString() // test with equal amounts (prev 0.2)
      const takerAssetAmount = web3.utils.toWei('0.3').toString()
      const expirationTimeSeconds = new BigNumber(
        Date.now() + 3600000
      ).toString() // Valid for up to an hour

      // default account sets allowance to 0x erc20proxy
      await baseContracts['WETH9'].deposit({
        value: makerAssetAmount,
        from: accounts[1]
      })
      await baseContracts['WETH9'].approve(erc20ProxyAddress, makerAssetAmount
        ).send({ from: accounts[1] })
      // account 0 must set allowance to totle token transfer proxy
      await baseContracts['RigoToken'].approve(tokenTransferProxy, makerAssetAmount)

      // Generate order
      const order = {
        makerAddress,
        takerAddress,
        feeRecipientAddress,
        senderAddress,
        makerAssetAmount,
        takerAssetAmount,
        makerFee,
        takerFee,
        expirationTimeSeconds,
        salt,
        makerAssetData,
        takerAssetData,
        exchangeAddress
      }
      const providerEngine = web3.currentProvider

      const signerAddress = accounts[0]

      const signedOrder = await signatureUtils.ecSignOrderAsync(
        providerEngine,
        order,
        signerAddress
      )

      const takerAssetFillAmount = (takerAssetAmount / 2).toString() // partial fill

      const encodedOrder = await web3.eth.abi.encodeParameters(
        [
          'tuple(address,address,address,address,uint256,uint256,uint256,uint256,uint256,uint256,bytes,bytes)',
          'uint256',
          'bytes'
        ],
        [
          [
            makerAddress,takerAddress,feeRecipientAddress,senderAddress,
            makerAssetAmount,takerAssetAmount,makerFee,takerFee,expirationTimeSeconds,salt,
            makerAssetData,takerAssetData
          ],
          takerAssetFillAmount,
          signedOrder.signature
        ]
      )

      const tradeId = '0xfa39c1a29cab1aa241b62c2fd067a6602a9893c2afe09aaea371609e11cbd92d' // mock id bytes32
      const isSell = true
      const optionalTrade = false
      const tokenAmount = takerAssetFillAmount
      const minimumExchangeRate = 10000000 // Allowable Price Change (%)
      const minimumAcceptableTokenAmount = 10000 // Minimum Token Fill Quantity(%)

      const feeAccount = accounts[2]
      /*
      // TODO: verify correctly initiated
      if(!affiliateRegistry.isValidAffiliate(feeAccount)){
          feeAccount = defaultFeeAccount;
      }
      */

      await expect(totlePrimaryInstance.methods.performRebalance(
        [
          [
            isSell,
            rigoTokenAddress,
            tokenAmount,
            optionalTrade,
            minimumExchangeRate, // check on value
            minimumAcceptableTokenAmount,
            [
                [
                  zeroExHandlerAddress,
                  encodedOrder
                ]
            ]
          ]
        ],
        feeAccount,
        tradeId
      ).send({ ...transactionDefault })
      ).rejects.toThrowErrorMatchingSnapshot()
    })
  })
})
