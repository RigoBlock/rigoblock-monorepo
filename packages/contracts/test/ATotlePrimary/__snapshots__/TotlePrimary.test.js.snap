// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` TotlePrimary performRebalance performs a 0x GRG buy order on totle 1`] = `"Returned error: VM Exception while processing transaction: revert"`;
